bigchaindb/tendermint/lib.py

```diff
@@ -326,7 +327,7 @@ def get_spent(self, txid, output, current_transactions=[]):
         if len(transactions) + len(current_spent_transactions) > 1:
             raise DoubleSpend('tx "{}" spends inputs twice'.format(txid))
         elif transactions:
-            transaction = Transaction.from_db(self, transactions[0])
+            transaction = operation_class(transactions[0]).from_db(self, transactions[0])
```

Comment:
To make the core transaction module independent from the plugin modules, we can do DI the following way:

```
class Transaction:
    type_registry = {}

    @classmethod
    def register_type(typ, clas):
        type_registry[typ] = clas

    @classmethod
    def from_whatever(tx):
        return cls(type_registry[tx['operation']])
```

Then, inside `a_plugin.py` one can do smth like:

```
from core import Transaction

Transaction.register_type('election', ValidatorElection)
```

What do you think?